course-content/30_bcdata.qmd

```diff
@@ -0,0 +1,233 @@
+---
+title: "Introduction to bcdata"
+teaching: 45
+exercises: 10
+questions:
+- "Fill this in!"
+objectives:
+- "Fill this in too!"
+keypoints:
+- "Same verse!"
+---
+
+<center>
+    <img src="images/bcdata/data_catalog.png" 
+    width="80%" style="margin:auto"/>
+    <p style="text-align: center">
+        B.C. Data Catalogue
+    </p>
+</center>
+
+The British Columbia government hosts over 2000 tabular and geospatial data sets in the [B.C. Data Catalogue](https://catalogue.data.gov.bc.ca/). Most data is available through the B.C. Data Catalogue under an open licence. The `bcdata` package provides a programming interface to the **geospatial** data sets of the  B.C. Data Catalogue. This allows Python users to import data directly from the B.C. Data Catalogue into their Python session. Through this functionality the bcdata package connects British Columbia government public data holdings in the B.C. Data Catalogue with the vast capabilities of Python.
+
+## Installing
+
+Unlike the rest of the packages we have downloaded (using `conda` commands), the `bcdata` package is not available for download via conda (this is often the case for smaller packages that are not widely distributed globally). This is one of those instances where we need to mix our use of `pip` and `conda`. However, the installation process proceeds in the same fashion as before. These commands can be done from an anaconda prompt, or from the command prompt (`cmd`) utility inside VS Code. 
+
+```{.bash filename='Anaconda Prompt'}
+> conda activate ds-env
```

Comment:
So on a mac, this didn't work out-of-the box for me. It errored installing a dependency (`psycopg2`). I did `conda install `psycopg2; pip install bcdata`. It's entirely possible it's just my ignorance of package managemnt in Python...

Reply:
Workflow for installing bcdata in VS Code was super clear and worked perfectly. Also very easy to work in Jupyter if folks do that, !pip install from a cell.

Reply:
@ateucher were you working out of a relatively clean conda environment at the time? (i.e. the ds-env one I have them setup at the beginning). It seems to me that it should be relatively OS agnostic, but I admit that package management is still a bit black magic to me sometimes as well...

Reply:
Yeah, I think pip was trying to build `psycopg2` from source and ran into issues, Conda had no issues, then `pip install bcdata` was fine. So hopefully you don't run into this in the course, but if you do hopefully this workaround is useful!